src/network/mcpe/NetworkSession.php

```diff
@@ -1163,6 +1169,25 @@ public function onOpenSignEditor(Vector3 $signPosition, bool $frontSide) : void{
 		$this->sendDataPacket(OpenSignPacket::create(BlockPosition::fromVector3($signPosition), $frontSide));
 	}
 
+	/**
+	 * @param EnchantmentOption[] $options
+	 */
+	public function sendEnchantOptions(array $options) : void{
+		$protocolOptions = [];
+		$optionId = 0;
+
+		foreach($options as $option){
+			$protocolEnchantments = array_map(
+				fn(EnchantmentInstance $e) => new Enchant(EnchantmentIdMap::getInstance()->toId($e->getType()), $e->getLevel()),
+				$option->getEnchantments()
+			);
+			$protocolOptions[] = new EnchantOption($option->getRequiredXpLevel(), $optionId, $protocolEnchantments, [], [], $option->getDisplayName(), $optionId);
```

Comment:
Is there any difference in behaviour if we do this as opposed to using all three arrays? I'm not saying it should e changed, just wondering if there's a behavioural difference to be aware of.

Might be worth adding a comment here to indicate why the other two arrays are empty.

Reply:
No, there is no any difference. [gophertunnel](https://github.com/Sandertv/gophertunnel/blob/381bd5bc31cf37b33eb2bcb06a3ff9968aee2965/minecraft/protocol/enchant.go#L38) has documented the following:

> Each array represents enchantments that will be added to the item with a different activation type. The arrays in which enchantments are sent by the vanilla server are as follows:
> * slice 1 { protection, fire protection, feather falling, blast protection, projectile protection, thorns, respiration, depth strider, aqua affinity, frost walker, soul speed }
> * slice 2 { sharpness, smite, bane of arthropods, fire aspect, looting, silk touch, unbreaking, fortune, flame, luck of the sea, impaling }
> * slice 3 { knockback, efficiency, power, punch, infinity, lure, mending, curse of binding, curse of vanishing, riptide, loyalty, channeling, multishot, piercing, quick charge }

Anyway, everything works fine without it (maybe it's only used in BDS), and I don't see the point of adding logic to separate enchantments across these three arrays.